registry/storage/paths.go

```diff
@@ -242,6 +246,18 @@ func pathFor(spec pathSpec) (string, error) {
 		return path.Join(append(repoPrefix, v.name, "_uploads", v.id, "hashstates", string(v.alg), offset)...), nil
 	case repositoriesRootPathSpec:
 		return path.Join(repoPrefix...), nil
+	case referrersLinkPathSpec:
+
+		revisionComponents, err := digestPathComponents(v.revision, false)
+		if err != nil {
+			return "", err
+		}
+
+		subjectComponents, err := digestPathComponents(v.subjectRevision, false)
+		if err != nil {
+			return "", err
+		}
+		return path.Join(append(append(append(append(repoPrefix, v.name, "_referrers", "subjects"), revisionComponents...), subjectComponents...), "link")...), nil
```

Comment:
4 x `append` do not seem right. Could you rewrite it to multiple lines?

Reply:
Rewrote it.